### DISCUSSION TITLE
php artisan test is only running latest created test and ignoring other ones.

### DISCUSSION THREAD

**Discussion by dnsnx** [laravel/framework#40758] [Q&A]
Hi, 

this is the first time, I use this discussion here, so please be patient. 
I'm having an issue with Laravel / PHPUnit Testing.
I'm using the latest Laravel version (8.82.0) and PHPUnit with version 9.5.13 on PHP 7.4.

I have created 2 units tests in my application, but only one test is being executed when I run `php artisan test`, when I use Laravel's `TestCase` class.

Here are my two tests:

**WoodWoodConnectionTest:**

```
<?php
namespace Tests\Unit;

//use PHPUnit\Framework\TestCase;
use Tests\TestCase;

class WoodWoodConnectionTest extends TestCase {
    public function test_wood_wood_calculation() {
        //Arrange
        $verificationDataOfHCW = new \App\Models\Nachweis\Daten\HCW();
        $verificationDataOfHCW->connection_type = \Config::get('constants.verbindungstyp.holzholz');
        ...
        $calculationRepository = \App::make('\App\Repositories\Nachweis\HCWRepositoryImpl');

        //Act
        $calculationRepository->berechnen($verificationDataOfHCW);

        //Assert
        $this->assertEquals(365, $verificationDataOfHCW->component_one_p_k);
        $this->assertEquals(28.80, $verificationDataOfHCW->f_v_0_rk);
        ...
    }
}
```

Here are my two tests:

**WoodConcreteConnectionTest:**

```
<?php
namespace Tests\Unit;

//use PHPUnit\Framework\TestCase;
use Tests\TestCase;

class WoodConcreteConnectionTest extends TestCase
{
    /**
     * A basic unit test example.
     *
     * @return void
     */
    public function test_wood_concrete_calculation()
    {
        //Arrange
        $verificationDataOfHCW = new \App\Models\Nachweis\Daten\HCW();
        $calculationRepository = \App::make('\App\Repositories\Nachweis\HCWRepositoryImpl');
        
        //Act
        //$calculationRepository->berechnen($verificationDataOfHCW);

        //Assert
        $this->assertTrue(true);
    }
}
```

When I run `php artisan test` it tells me that the test `WoodConcreteConnectionTest` is being successful, but the other test is not running. 
If I create a new test with `php artisan make:test TestTest`, the test will be executed too (that means 2 tests are running successful), but when I change 

`use PHPUnit\Framework\TestCase;`

to

`use Tests\TestCase;`

only the `TestTest` is being executed. 
I already put the tests directly in the `Unit` folder without subfolders etc. 
I checked the `phpunit.xml`, `TestCase` class and `CreatesApplication` trait (copied it from the Laravel project source).

I described the issue on StackOverflow, too (https://stackoverflow.com/questions/70918132/laravel-testing-only-one-test-is-being-executed-when-running-command-php-arti).

I'm not sure if it is an issue of Laravel or PHPUnit, but when I'm using the PHPUnit's `TestCase` class more than one test is being executed. So I assume it is more a Laravel issue. 
Are there any other locations in a Laravel project which controls or have influence on testing?
Any help appreciated! Many thanks in advance!

**Comment by gguney** [comment]
Did you have any "dump" function running inside your code? If so please remove the dump function inside your code. I too faced similar issue that dump function stops the tests running without a response.

**Comment by dnsnx** [comment]
I'm not 100% sure what do you mean with "dump" function? The issue is, when I ran each test individually, all tests are successful. 
So I don't think, that the tests run into a failure.

**Reply by gguney** [reply]
Can you share phpunit.xml ? Maybe there is something missing in there

**Reply by dnsnx** [reply]
Yes, of course. I already tried here several versions, too, but all with same result. 
This is the one I'm using now:

```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
         bootstrap="vendor/autoload.php"
         colors="true"
>
    <testsuites>
        <testsuite name="Unit">
            <directory suffix="Test.php">./tests/Unit</directory>
        </testsuite>
        <testsuite name="Feature">
            <directory suffix="Test.php">./tests/Feature</directory>
        </testsuite>
    </testsuites>
    <filter>
        <whitelist processUncoveredFilesFromWhitelist="true">
            <directory suffix=".php">./app</directory>
        </whitelist>
    </filter>
    <php>
        <server name="APP_ENV" value="testing"/>
        <server name="BCRYPT_ROUNDS" value="4"/>
        <server name="CACHE_DRIVER" value="array"/>
        <server name="MAIL_MAILER" value="array"/>
        <server name="QUEUE_CONNECTION" value="sync"/>
        <server name="SESSION_DRIVER" value="array"/>
        <server name="TELESCOPE_ENABLED" value="false"/>
    </php>
</phpunit>
```

**Reply by gguney** [reply]
Can you add below code to your phpunit tag?

```
    convertErrorsToExceptions="true"
    convertNoticesToExceptions="true"
    convertWarningsToExceptions="true"
    stopOnFailure="false"
```

Like this:

```
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          convertErrorsToExceptions="true"
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
          stopOnFailure="false"
          xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
          bootstrap="vendor/autoload.php"
          colors="true"
>


```

and try again?

**Reply by dnsnx** [reply]
I've added it as you proposed, but the result is the same. 
Only one test is being executed, when running command `php artisan test`. 

```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
        convertErrorsToExceptions="true"
        convertNoticesToExceptions="true"
        convertWarningsToExceptions="true"
        stopOnFailure="false"
        bootstrap="vendor/autoload.php"
        colors="true"
>
....
```

Also no errors or hints that something went wrong...

**Reply by gguney** [reply]
Can you comment-out or remove all the codes inside your test and just return true to see if there is something wrong with the code or the test

**Reply by gguney** [reply]
![Screen Shot 2022-02-23 at 09 51 03](https://user-images.githubusercontent.com/16459896/155273048-d3c73f40-fd69-45a0-ba52-1f18e528b85b.png)

`<?php

namespace Tests\Unit;

use Tests\TestCase;

class WoodConcreteConnectionTest extends TestCase
{
    /**
     * A basic unit test example.
     *
     * @return void
     */
    public function test_wood_concrete_calculation()
    {
        $this->assertTrue(true);
    }
}
`

I tried like this with default Laravel 9 and result was expected and run all tests

**Reply by gguney** [reply]
Do you have dd or somekind of stopper in your code? Because if I put dd in my code that tests used, test stop without giving any clue.
![Screen Shot 2022-02-23 at 09 55 48](https://user-images.githubusercontent.com/16459896/155273551-c3dbb4f6-def5-46dd-a7b0-14a7cca46349.png)


As you can in the first part I added dd to my code and test stopped without notice.

**Reply by dnsnx** [reply]
It's really crazy. I've commented out all code and just do

`$this->assertTrue(true);`

but the result is this:

![2022-02-23 07_55_00-WoodConcreteCenterOfPanelConnectionTest php-Visual Studio Code](https://user-images.githubusercontent.com/20823625/155273638-a18965fc-6f2c-4fdf-a5dc-899333a9e119.png)

So as you can see, it only one test is done.
There are four tests, which should be executed.

I'm using the latest version Laravel 8. As I wrote in my original post, the project was created in Laravel 5.4 und upgraded step by step to the current version. Could it be, that there is an issue with this? So maybe, is there another location in the Laravel project, which controls testing in any way? I just don't understand it and am thinking about possible reasons.

**Reply by dnsnx** [reply]
Actually I don't think there is a stopper. The function I'm testing is just calculating the verification. There is no ouput in it. And then the test should not run successfully, when I run it individually, shouldn't it?

**Reply by gguney** [reply]
Ah you may be right, maybe you missed something when upgrading, if it is working before in Laravel 5.4 then there can be a problem the upgrade process. 

If there is some kind of stopper in your code then it runs but then stops silently. However, since you removed all the code in your test and it still stops, I think there is another error with the upgrade process.

What is the next Test? I mean WoodConcreteCenterOfPanelConnectionTest is first then which test is coming? Can you check that test too remove all of its content and add assertTrue?

**Reply by dnsnx** [reply]
I can't say whether the test already worked in 5.4. These tests didn't exist then. 
Do you have a guess as to where the error could be located?
I went through all the upgrade guides a few days ago and couldn't find any place where there had been significant changes to testing.

**Reply by gguney** [reply]
Can you use --filter option to jump over these test and run new tests. I've never encountered something like that before and can't say anything without seeing it. 

Can you try to run your codes with `./vendor/bin/phpunit` ? not php artisan test directly

**Reply by dnsnx** [reply]
I already tried this, too. Same result. No change.

**Reply by gguney** [reply]
I do not have any other idea, sorry. If I face similar issue I'll let you know

**Reply by dnsnx** [reply]
Yeah, but many thanks to you for your help!

**Reply by gguney** [reply]
Can you check your copy of use Tests\TestCase; ?

**Reply by dnsnx** [reply]
I've copied the files from the repo.

TestCase.php:
```
<?php
namespace Tests;

use Illuminate\Foundation\Testing\TestCase as BaseTestCase;

abstract class TestCase extends BaseTestCase
{
    use CreatesApplication;
}
```

CreatesApplication.php:
```
<?php
namespace Tests;
use Illuminate\Contracts\Console\Kernel;

trait CreatesApplication
{
    /**
     * Creates the application.
     *
     * @return \Illuminate\Foundation\Application
     */
    public function createApplication()
    {
        $app = require __DIR__.'/../bootstrap/app.php';

        $app->make(Kernel::class)->bootstrap();

        return $app;
    }
}
```
